src/common/buffer.cc

```diff
@@ -2278,6 +2278,26 @@ int buffer::list::write_fd_zero_copy(int fd) const
   return 0;
 }
 
+__u32 buffer::list::crc32c(unsigned off, size_t len, __u32 crc) const
+{
+  if (off == 0 && len == length()) {
+    return crc32c(crc); //fall back to full implementation that supports caching.
+  }
+  buffer::list::const_iterator it = begin();
+  it.advance(MIN(off, length()));
```

Comment:
This advance() call is not free--it has to traverse the bufferlist's list of ptr's.  I was imagining a crc32c method that works directly on the iterator and uses get_ptr_and_advance().